lib/iterator_range.h

```diff
@@ -0,0 +1,57 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef LIB_ITERATOR_RANGE_H_
+#define LIB_ITERATOR_RANGE_H_
+
+#include <utility>
+
+namespace Util {
```

Comment:
Is this a partial substitute for the lack of C++20 ranges? Can you add a comment describing its purpose? I wonder how transient this utility library will be. 

Alternatively, we could add a drop-in substitute like https://github.com/ericniebler/range-v3 to the compiler and then replace it once we migrate to C++20.

Reply:
I guess what I am getting at is that these little libraries end up hanging around and over the years people lose information about what they are for. With the vendored dependencies it is a little more explicit because you know they are placeholders until the standards progress. 

Can we add an explanation why this was added, what it is being used for and how it could be replaced, if at all?

Reply:
Yes, sure. And remember: "every large project should have lisp interpreter inside" :)